### PR TITLE
Bmf refactor ansible scripts

### PR DESCRIPTION
**Pull request by brandon-fryslie**
Refactor scripts, add etcher-cli so we can flash the card and configure SSH + WiFi in one command, add rad-shell to the development environment

### CODE REVIEW

File: ansible/roles/set-hostname/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    content: "{{ new_hostname }}"
+    dest: /etc/hostname
+  register: hostname_changed
+
+- name: Update /etc/hosts
+  become: yes
+  when: new_hostname is not undefined
+  lineinfile:
+    path: /etc/hosts
+    regexp: '127.0.1.1	raspberrypi'
```
Comment (zenblender) [inline ansible/roles/set-hostname/tasks/main.yml:16]: Is this a regex string? If so, should it be something like `127\.0\.1\.1\s+raspberrypi`?
Also, is this supposed to be 127.0.0.1 or 127.0.1.1?

File: zsh-plugins/boomwaffle-pi/boomwaffle-pi.zsh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+export PATH="${PATH}:~/code/boom-waffle/bin"
+
+# Add your boomwaffle ssh key to the ssh agent
+boomwaffle-ssh() {
+  eval $(~/code/boom-waffle/bin/add-ssh.sh)
+}
```
Comment (zenblender) [inline zsh-plugins/boomwaffle-pi/boomwaffle-pi.zsh:7]: Ah interesting. Does this assume my local dev is using `~/code/boom-waffle` as my dir? Mine is somewhere else but I can move it :)

File: bin/0-boot-to-wifi.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  green "Finished flashing SD card"
+  diskutil mountDisk "${disk_to_flash}"
+}
+
+cd $script_path/../ansible
+
+yellow "Running playbook boot-to-wifi.yml"
+
+ansible-playbook -i '127.0.0.1,' --extra-vars "ssid='${ssid}' wifi_password='${wifi_password}'" boot-to-wifi.yml
```
Comment (zenblender) [inline bin/0-boot-to-wifi.sh:116]: Ah ha, does this now properly deal with wifi SSIDs / passwords that contain spaces?

File: bin/1-first-boot.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+# Wait for raspberri pi to start up
+wait_for_pi() {
+  local hostname=$1
+  set +e
+  local counter=0
+  local max_tries=60
+
+  yellow "Waiting for pi to start up at hostname: ${hostname}"
```
Comment (zenblender) [inline bin/1-first-boot.sh:26]: Wow, so much more error checking now! Cool!

File: zsh-plugins/boomwaffle-pi/boomwaffle-pi.zsh (hunk 2 above)
Comment (brandon-fryslie) [reply zsh-plugins/boomwaffle-pi/boomwaffle-pi.zsh:7]: we can make this an environment variable.  this script is still in the early stages

File: ansible/roles/set-hostname/tasks/main.yml (hunk 1 above)
Comment (brandon-fryslie) [reply ansible/roles/set-hostname/tasks/main.yml:16]: it does work the way it is, its copied directly from the pi itself.  if you leave this out, there isn't any practical effect except it throws a warning when you use sudo.  it probably could be changes to the regex you've got, try it out and see if it works :)

File: bin/0-boot-to-wifi.sh (hunk 3 above)
Comment (brandon-fryslie) [reply bin/0-boot-to-wifi.sh:116]: yep!  :) it needed the single quotes: `wifi_password='${wifi_password}'`

**Pull request merged**